src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +19,53 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return ampHourUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if(currCurrent_>=0)
+    {
+        return false;
+    }
+    return true;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double ampHoursLeft;
+    int current = currCurrent_;
+    if(isCharging())
+    {
+        ampHoursLeft = BATTERY_AMP_HOUR_CAPACITY - ampHourTotal_;
+        current *= -1;
+    }
+    else
+    {
+        ampHoursLeft = ampHourTotal_;
+    }
+    double msecLeft = (ampHoursLeft * HOURS_TO_MSECS) / current;
+    QTime QTimeLeft = QTime(0,0,0,0).addMSecs(msecLeft);
```

Comment:
It's not necessary to name variables with their type, doing this is called hungarian notation and generally that's not how we name things on the team and is frowned upon. `timeLeft` should suffice :)